### PR TITLE
Move Vulkan HAL flags into vulkan_driver_module.cc.

### PR DESCRIPTION
**Pull request by ScottTodd**
Helps with https://github.com/google/iree/issues/898 (absl::flags contributes ~38KB to our binaries)

These options are now no longer configurable from the [Vulkan HAL C API](https://github.com/google/iree/blob/main/iree/hal/vulkan/api.h). We can add them there if needed.

### CODE REVIEW

File: iree/hal/vulkan/vulkan_device.cc [hunk 1]
```diff
@@ -227,6 +224,7 @@ StatusOr<ref_ptr<VulkanDevice>> VulkanDevice::Create(
     VkPhysicalDevice physical_device,
     const ExtensibilitySpec& extensibility_spec,
     const ref_ptr<DynamicSymbols>& syms,
+    bool force_timeline_semaphore_emulation,
```
Comment (benvanik) [inline iree/hal/vulkan/vulkan_device.cc:227]: we'll want to have an options struct for this kind of stuff - probably worth doing now so we don't grow more later

Comment (benvanik) [inline iree/hal/vulkan/vulkan_device.cc:227]: move after extensibility_spec to match the order used elsewhere and keep syms at the end

File: iree/hal/vulkan/vma_allocator.cc [hunk 2]
```diff
@@ -23,6 +22,7 @@
 #include "iree/hal/vulkan/vma_buffer.h"
 
 #if VMA_RECORDING_ENABLED
```
Comment (benvanik) [inline iree/hal/vulkan/vma_allocator.cc:24]: (as you mention) it'd be good follow-up to make this options on the creation instead so we can remove this (and compile out the options in release builds/etc)

**Review by benvanik — CHANGES_REQUESTED** [review]

File: iree/hal/vulkan/vulkan_device.cc (hunk 1 above)
Comment (ScottTodd) [reply iree/hal/vulkan/vulkan_device.cc:227]: I was going to wait until a second option like this was added, but ok :P

File: iree/hal/vulkan/vma_allocator.cc (hunk 2 above)
Comment (ScottTodd) [reply iree/hal/vulkan/vma_allocator.cc]: Done in this PR. PTAL.

**Review by benvanik — APPROVED** [review]

**Pull request merged**